implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -58,6 +58,16 @@ protected SmallRyeConfig(List<ConfigSource> configSources, Map<Type, Converter>
                 return convert(value, aClass);
             }
         }
+
+        // check for  Optional numerical types to return their empty()
+        // if the property is not found
+        if (aClass.isAssignableFrom(OptionalInt.class)) {
+            return (T)OptionalInt.empty();
```

Comment:
This could be `aClass.cast(OptionalInt.empty())` and that would make it a safe cast.

Reply:
fixed, thanks